lib/command.js

```diff
@@ -1054,11 +1054,16 @@ Expecting one of '${allowedValues.join("', '")}'`);
     const subCommand = this._findCommand(commandName);
     if (!subCommand) this.help({ error: true });
 
-    if (subCommand._executableHandler) {
-      this._executeSubCommand(subCommand, operands.concat(unknown));
-    } else {
-      return subCommand._parseCommand(operands, unknown);
-    }
+    let actionResult;
+    actionResult = this._chainOrCallHooks(actionResult, 'preSubCommand');
```

Comment:
I think this will call the `preSubCommand` for all parent commands as well. So when there are nested commands, the same hook will get called multiple times.

Reply:
This could perhaps get handled by the callback looking at the parameters to tell what the context is, but my first thought is that the callback is only called once at each level by the current command before its subcommand is called. I'll need to think about that. The existing hooks only get called from one location and I haven't thought through this new pattern.